test/functional/misc/vector_api_cli_test.go

```diff
@@ -36,5 +35,19 @@ var _ = Describe("[Functional][Misc][API_CLI] Functional test", func() {
 			out, _ := framework.RunCommand(constants.CollectorName, `curl`, `-sv`, `-m`, `5`, `--connect-timeout`, `3`, `http://127.0.0.1:8686/health`)
 			Expect(out).To(ContainSubstring(`{"ok":true}`))
 		})
+		It("should remap Proxy env variable from lowercase to uppercase if it's set", func() {
+			envVisitor := func(b *runtime.PodBuilder) error {
+				b.GetContainer(constants.CollectorName).AddEnvVar("http_proxy", "http://my_proxy:8080")
+				b.GetContainer(constants.CollectorName).AddEnvVar("https_proxy", "https://my_proxy:9090")
+				b.GetContainer(constants.CollectorName).AddEnvVar("no_proxy", "my-host,your-host")
+				return nil
+			}
+			Expect(framework.DeployWithVisitor(envVisitor)).To(BeNil())
+			out, err := framework.ReadCollectorLogs()
+			Expect(err).To(BeNil())
+			Expect(out).To(ContainSubstring(`HTTP_PROXY=http://my_proxy:8080`))
```

Comment:
I believe a more accurate assertion would be to exec into the  pod, get the env vars, and check explicitly for the environment variable and value

Reply:
yes, get the env vars will be better but with here var sets only in the process running that script, It does not persist globally in the container environment.